doc/source/development/rfc/rfc87_signed_int8.rst

```diff
@@ -0,0 +1,119 @@
+
+- drivers that need to support writing signed 8-bit integers must provide a
+  creation option for that purpose (generally called PIXELTYPE=SIGNEDBYTE
+  itself).
+
+- but more importantly, the absence of a proper data type means that it is easy
+  to forget to test the PIXELTYPE=SIGNEDBYTE metadata item in all places where
+  the value of pixels is taken into account. There were special cases for
+  statistics computations, but most of the other code, such as the overview or
+  warping computation code had no provision for it, and consequently
+  mis-interpreted negative values in the range [-128,-1] as positive values in
+  the range [128,255].
+
+Details
+-------
+
+A new ``GDT_Int8`` = 14 item is added to the :cpp:enum:`GDALDataType` enumeration.
+
+It is taken into account in all places in gcore/ and alg/ that have specific
+behavior depending on the value of GDALDataType. All places that only dealt
+specifically with a subset of types (e.g Byte, Float32) and promoted other
+types to the closest one will automatically have proper support for the new
+data type (e.g overview computation).
+
+Existing drivers that, on reading, reported GDT_Byte + PIXELTYPE=SIGNEDBYTE are
+modified to report GDT_Int8. This has backward compatibility impact to external
+code. This affects the EEDAI, ERS, GTA, GTiff, HFA, netCDF, PostGISRaster, EHDR,
+RRaster, Rasterlite2 and Zarr drivers.
+
+Existing drivers that, on writing, accepted the PIXELTYPE=SIGNEDBYTE, are modified
+to accept GDT_Int8 as a valid data type in their Create()/CreateCopy() implementations,
+and advertize "GDT_Int8" in their GDAL_DMD_CREATIONDATATYPES metadata item.
+The PIXELTYPE=SIGNEDBYTE creation option is kept but deprecated and discouraged:
+working with it prior to this RFC was clunky at times, and will remain such.
+
+The MEM, HDF4, HDF5, KEA and PDS4 drivers are extended to support GDT_Int8 on reading
+(and writing, except HDF5 that is read-only).
+
+Bindings
+--------
+
+The GDT_Int8 constant is made available in SWIG bindings.
+
+The numpy inteface of the Python bindings is modified to map GDT_Int8 to
+``numpy.int8``.
+
+Backward compatibility
+----------------------
+
+The PIXELTYPE=SIGNEDBYTE band metadata item in IMAGE_STRUCTURE is no longer
+reported. Code that depended on it needs to switch testing for GDT_Int8.
+Code that has "switch"-like constructs on GDALDataType must also account for
+GDT_Int8.
+
+Documentation
+-------------
+
+Documentation of utilities that have a ``-ot`` switch is modified to mention Int8.
+
+:ref:`rfc-14` will be amended to point to this RFC, and mention that starting
+with GDAL 3.7.0, PIXELTYPE=SIGNEDBYTE is no longer the technical solution.
+
+Testing
+-------
+
+The test_gdal.cpp and testcopywords.cpp test files are extended to test functions
+like :cpp:func:`GDALDataTypeIsInteger`, :cpp:func:`GDALDataTypeIsFloating`,
+:cpp:func:`GDALDataTypeIsComplex`, etc. and all data type conversion combinations
+from/into GInt8.
```

Comment:
Is it worth to touch on the conversions here in the RFC? e.g. lossy, linear mapping, source min/max -> target min/max, what about preserving signed Zero when converting to unsigned channel type, etc. implications, and also on mixed channel arithmetic, if any.

Reply:
not completely sure to follow all what you've in mind here, but the conversion rules applied for Int8 are consistent with the ones applied for similar types such as Int16 or Int32.

Reply:
That will do 😄